client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
```

Comment:
> Is that designed to wrap a generic actionCreator and add 'stat-bumping' abilities to a fired action?
> I would like to see usage of if in mapActionsToProps :)

yes - because I like composition by aggregation over inheritance and imperative calls :wink:

I will add this into the README, but here is how it would work

```js
import {
	withAnalytics,
	bumpStat
} from 'state/analytics/actions';

import { setWidget } from 'widgets/actions';

const mapDispatchToProps = dispatch => ( {
	selectWidget: widget => dispatch( withAnalytics(
		bumpStats( 'widget_selected', widget.name ),
		widget
	) )
} );
```